src/Squidex/app/shared/services/schemas.service.ts

```diff
@@ -99,19 +100,24 @@ export class SchemaDetailsDto extends SchemaDto {
         super(links, id, name, category, properties, isSingleton, isPublished, created, createdBy, lastModified, lastModifiedBy, version);
 
         if (fields) {
-            let listFields = this.fields.filter(x => x.properties.isListField && x.properties.isContentField);
+            this.listFields = this.getField(x => x.properties.isListField);
+            this.listFieldsEditable = this.listFields.filter(x => x.isInlineEditable);
 
-            if (listFields.length === 0 && this.fields.length > 0) {
-                listFields = [this.fields[0]];
-            }
+            this.referenceFields = this.getField(x => x.properties.isReferenceField);
+        }
+    }
 
-            if (listFields.length === 0) {
-                listFields = NONE_FIELDS;
-            }
+    private getField(predicate: (field: RootFieldDto) => boolean) {
+        let fields = this.fields.filter(x => predicate(x) && x.properties.isContentField);
 
-            this.listFields = listFields;
-            this.listFieldsEditable = listFields.filter(x => x.isInlineEditable);
+        if (fields.length === 0 && this.fields.length > 0) {
+            fields = [this.fields[0]];
```

Comment:
Is there any chance this can be null instead of 0? If so what happens?

Reply:
No chance.